### PR TITLE
p0fv3 support - tcp/http/mtu passive fingerprinting

### PR DESCRIPTION
**Pull request by Nisitay**
This adds support to `p0fv3`

Some of the new features and changes:
* Parsing of p0f.fp using the p0fv3 formatting
* Complete IPv4/IPv6/TCP/HTTP parsing
*  TCP/HTTP/MTU passive fingerprinting
* Changed the name of the old `p0f` module to `p0fv2` to allow support for v2

### CODE REVIEW

**Comment by gpotter2** [comment]
This looks like a really good start ! Especially condisering p0f support has been broken for quite some time

**Comment by Nisitay** [comment]
Thanks! After doing several tests and checking that the main functions work, I think this is now ready for review

**Comment by Nisitay** [comment]
I've changed the structure of the signatures and records to further improve readability. I won't change anything else for now, so I'm just waiting for the review.

File: scapy/modules/p0f.py [hunk 1]
```diff
@@ -1,128 +1,578 @@
-# This file is part of Scapy
-# See http://www.secdev.org/projects/scapy for more information
-# Copyright (C) Philippe Biondi <phil@secdev.org>
-# This program is published under a GPLv2 license
```
Comment (gpotter2) [inline scapy/modules/p0f.py:4]: Could you keep a header + docstring?

**Review by gpotter2 — COMMENTED** [review]
This looks really good. Could you add a test that first impersonates then checks that the impersonated packet is properly detected? Would be great thanks !

Sorry for the delay

**Comment by Nisitay** [comment]
@gpotter2 I've added the `p0f_impersonate` function and some tests. The function only supports TCP (SYN/SYN+ACK) packets. I thought about implementing a similar function for HTTP, but since the HTTP fingerprints are based on substrings found in the HTTP headers, I think it would be very hard to generate headers (or change existing headers) while keeping the HTTP headers functional.

**Comment by gpotter2** [comment]
Thanks for the very cool PR !

**Pull request merged**